runtimes/runtimes/lsp/router/loggingServer.ts

```diff
@@ -0,0 +1,44 @@
+import { InitializeParams, InitializeResult, Logging } from '../../../server-interface'
+import { LspServer } from './lspServer'
+import { Connection } from 'vscode-languageserver/node'
+import { Encoding } from '../../encoding'
+import { DEFAULT_LOG_LEVEL, isValidLogLevel, LoggingImplementation, LogLevel } from '../../util/loggingUtil'
+
+export class LoggingServer {
+    private logger: Logging
+    private lspServer: LspServer
+
+    constructor(
+        private lspConnection: Connection,
+        private encoding: Encoding
+    ) {
+        this.logger = new LoggingImplementation(DEFAULT_LOG_LEVEL as LogLevel, this.lspConnection)
+        this.lspServer = new LspServer(this.lspConnection, this.encoding, this.logger)
+        this.lspServer.setInitializeHandler(async (params: InitializeParams): Promise<InitializeResult> => {
+            this.updateLoggingLevel(params.initializationOptions?.logLevel ?? ('log' as LogLevel))
```

Comment:
Do we have `logLevel` defined in `initializationOptions` in protocol?

Reply:
we don't need to define the same explicitly inside `initializationOptions` as it contains a map of string to any. Reference: [link](https://github.com/aws/language-server-runtimes/blob/b8e3adeff5a8396afc40062439b9541512a0d29d/runtimes/protocol/lsp.ts#L116)

Reply:
It's good to have it explicitly listed and/or documented. Otherwise, how will clients know what is possible?

Reply:
I'll update with the details under README.md

Reply:
As I mentioned in another comment, I think we need to explicitly add `logLevel` to protocol, since that's how we define contract with destination clients. It has to be added explicitly, as otherwise we may accidentally break protocol without noticing